src/value/builders.rs

```diff
@@ -11,7 +11,7 @@ use crate::Symbol;
 ///     .push("foo")
 ///     .build()
 ///     .into();
-/// let expected = Element::parse(r#"[1, true, "foo"]"#).unwrap();
+/// let expected = Element::read_one(r#"[1, true, "foo"]"#).unwrap();
```

Comment:
🗺️ In the previous PR, we introduced the method `Element::parse(ion_data) -> IonResult<Element>`.

The new `ElementReader` trait has three `read_` methods:
* `read_next_element()`
* `read_one_element()`
* `read_all_elements()`

In this PR, I added analogous convenience methods to `Element`:
* `Element::read_first()` -- (It's stateless, so you can't read more than the first one.)
* `Element::read_one()` -- This was previously called `parse`. It enforces that the input has exactly one value.
* `Element::read_all()`

A lot of the changes that follow are renaming `Element::parse` to `Element::read_one`.